### DISCUSSION TITLE
"hot reload"

### DISCUSSION THREAD

**Discussion by haringsrob** [livewire/livewire#4953] [3. Feature ideas]
Hey 👋 ,

I think it might be cool to add a "hot reload" method.

While this for sure is a bit challenging it might work by adding a filesystem watcher (blade + app). And on change, it would notify all components to force a redraw.

The front-end needs a listener that notifies the back-end of an fs update.

Have there been ideas around this in the past?

Might look into this at some point myself.

**Comment by joshhanley** [comment]
@haringsrob you can use browsersync, take a little configuring, but works great when setup.

I don't think being built into Livewire would work as there is no build process, and as such no filesystem watcher. 

My preference would be just to use existing packages like browsersync for it.

How would you imagine it working?

**Comment by haringsrob** [comment]
Hey @joshhanley,

My idea was to have a core listener on every component (only active in debug mode), called "forceRedraw".

Then on the front-end, there is a listener like is done with vue hot reload, but instead of reloading the CSS/js, it triggers the "forceRedraw" event `$emit('forceRedraw')` causing all the currently active components to refresh.

This is different from a browser refresh as the state should remain. And is very useful for working on more complex livewire components.

Cheers

**Reply by haringsrob** [reply]
I will give this a quick try today see if I can make a poc work and submit a pr for it.

**Comment by haringsrob** [comment]
Oh well, that was easier than expected.

It reuses the same watcher as Laravel mix --hot for this. I feel this is fine and needs no more complexity.

![Screen Recording 2022-05-12 at 08 52 10](https://user-images.githubusercontent.com/866743/168009710-3c4bff72-4a1b-47b6-bb72-2edf0300f9cd.gif)

**Comment by haringsrob** [comment]
Added a pr here: https://github.com/livewire/livewire/pull/4955
